### DISCUSSION TITLE
2023-10-20 - Testing Review - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#6982] [Chewing the Fat]
Hi Testers!

After completing the testing series, what did you learn?

https://www.ssw.com.au/rules/rules-to-better-testing/

**Comment by 2fernandez** [comment]
I learnt that UX Testing and Developer Testing are very similar, at least in principle. 
The higher the risk, the greater the testing required.
 UX Testing also approaches testing from a Psychological angle. 
Fix the underlying psychology, and you typically fix more significant issues.

**Comment by brydeno** [comment]
Testing is not about making software perfect, it's about making the risk of those imperfections causing problems lower.

**Comment by AlexRothwell** [comment]
I've learnt that a combination of the various types of testing is important to reduce risk in a product. This helps guarantee that it can provide the intended value

It's important however to be pragmatic with the testing; there's only so much time available and the product is pointless if it never gets to the user

**Comment by tkapa** [comment]
Preventing regression and testing your code is essential. But there's a point where is can become too much. Ask @KahaMason about the One... (million!) unit tests he's been dealing with over the past 2 years.

**Comment by ozdemir-mehmet** [comment]
As above I like being able to make code changes knowing the unit tests have my back.

**Comment by MarcusIrmscherSSW** [comment]
Great video - very entertaining and informative. Awesome work Luke and Piers. 🏆

**Comment by matt-goldman** [comment]
This was a great series and I really enjoyed the summary video. I think the next steps for SSW are to translate this into practical, hands-on guidance, and build this into our processes.

**Comment by YalalCakirSSW** [comment]
I've learned that I need to learn more about testing and its practices. Awesome content and video. I am going to watch the whole series. 😊

**Comment by zacharykeeping** [comment]
Good video - I enjoy these breakdowns and the testing series has been really informative and helpful overall. Mitigating risk through a wide range of tests is important and something I'll want to focus more on going forward.

**Comment by joshbermanssw** [comment]
I agree with Luke - going over and directly testing what someone has just built is a great form.

**Comment by cvionnet** [comment]
The CTF testing series was really valuable, and I hope to see more such series in the future.
The video is excellent; the duo between Piers and Luke works wonderfully.
I'm looking forward to watch the extended version.

**Comment by bahjatalaadelssw** [comment]
This DTF format is good to have a review and it's good to see devs giving insights

**Comment by ricksu978** [comment]
It's impossible to test everything, so it's important to prioritize testing on areas that are high risk and high value.

**Comment by hveraus** [comment]
Great content, and new presenting style, it is good to see 2 presenters, the dynamic between Luke and Piers created a more engaging presentation.

**Comment by Ben0189** [comment]
Testing series is awesome, it taught me different types of testing, when and why testing is necessary.

**Comment by UlyssesMaclaren** [comment]
best intro ever!

**Comment by AntPolkanov** [comment]
This video was a real morning mood lifter! 😄

Thanks for the series! We've learned quite a bit of theory, what's next? Wouldn't it be nice to share our experience and approaches to testing?

**Comment by jeoffreyfischer** [comment]
Good reminder to enhance the importance of testing when working on a project, and to communicate that to the client.

**Comment by AttackOnMorty** [comment]
Testing can improve the effectiveness of finding, locating, and fixing problems. Theoretically, the latter three are not fundamental complexity issues. But in practice, they exist in large numbers and even occupy more than half of the effective working time.

**Comment by Marxoz** [comment]
I loved the series and think it was really valuable for all devs, whether you were across testing practices before or not

**Comment by Levijj22** [comment]
I see high value in ensuring our devs have a good understanding of different test methods.

**Comment by jakebayliss** [comment]
Another awesome video by Luke and Piers, really great to go though the feedback on this series. Learned at lot about how much testing we should be doing, when to do more, when to do less, etc

**Comment by apforsyth** [comment]
Congrats Piers and Luke on a great presentation, it was valuable and enjoyable to watch.

**Comment by sethdaily** [comment]
Sounds like testing is important and not worth skimping on. If this results in more attention to better testing, that's a win.

**Comment by CoffeeBugHub** [comment]
I have learned a different variety of testing, and I hope there are more rules and best practices for testing internal projects. Btw, I love the video, it is very easy to follow and it is fun.

**Comment by vladislav-kir** [comment]
Testing series was great. It taught me more about different types of testing.

**Comment by danielmackay** [comment]
It's helped to identify extra levels of the testing pyramid that I normally wouldn't think of (e.g. Component Testing).

Should we have a repository that showcases the best testing practices across all types of testing?  We have some of this in the Clean Architecture template, but are missing things like Mutation Tests and Perf tests.

**Comment by amankumarrr** [comment]
I acquired knowledge about various testing methodologies and gained insights into their importance.

**Comment by ncn-ssw** [comment]
Though testing is not a new or glitzy topic, it deserves regular review to reinforce its importance and cover any incremental improvements in methodology.

**Comment by jackreimers** [comment]
Great video, it was good to get a high level summary of the results and the Luke and Piers presented them in a way which was very engaging.

**Comment by chrisschultzssw** [comment]
Great video, and I do feel like I know a lot more about testing after this series!

**Comment by tomek-i** [comment]
I found that visual regression testing saves a lot of pain when working with web and designers :-D

SB also integrates very well with [Figma](https://storybook.js.org/docs/react/sharing/design-integrations) and if you [publish the storybook](https://storybook.js.org/docs/react/sharing/publish-storybook) you can develop components and show case them to the designer / client without building out a page / workflow.
And you can do similar things like PR's and Code Reviews just for design :-D 


https://storybook.js.org/docs/react/writing-tests/visual-testing
https://www.chromatic.com/

**Comment by tombui99** [comment]
The testing series have been highly valuable, and I would love to see more of them. Rigorous testing is imperative in guaranteeing the flawless functionality and comprehensive completion of a project, and it makes sure to devoid of any errors or shortcomings.

**Comment by AshAnil** [comment]
Learned different types of testing. Great video ⭐

**Comment by Dhruv-0987** [comment]
Loved the video, 2 hosts are definitely better then one. My main takeaway was making testing compulsory for every PBI as part of the definition of done and everyone in the team should enforce it this takes more time but will ensure less time to repair it later.

**Comment by steven0x51** [comment]
Great video! Nice to hear the status after a series. And I have learnt the test pyramid.

**Comment by JeanThirion** [comment]
First of all - DTF FTW ⭐

Testing series was one of the best we had this year, very important and interesting topic. There are so many misconceptions about software testing, and I think we need to be stronger on that topic. 

Having a testing officer, someone like Lee, going around projects and making sure we have a coherent testing strategy for each big projects is I think a great idea. 

Very good series

**Comment by NadeekaJK** [comment]
Good to see the summary of the testing series.
I've worked with most of the layers of the testing pyramid in different projects, from this series I learned the term testing pyramid and how its all looks when put into perspective.

**Comment by Harry-Ross** [comment]
Really great to see the results of the testing CTF series in a concise video, was great to have a high-level overview of everyone's responses in the company. Also, testing is a really important topic that we should be keen to focus more on.

**Comment by Geordie88** [comment]
It was great to see the conversation summarized and I enjoyed the call outs of some of the spicier takes (brady). Testing is such an important step of any process and I'm glad it got a highlight :)

**Comment by MattParkerDev** [comment]
The Testing Review series was insightful, offering valuable lessons. It highlighted the importance of balancing comprehensive testing with project practicality. Prioritizing high-risk and high-value areas is key, as we can't test everything. The concept of the testing pyramid provides a structured approach to testing for efficiency. Overall, this series underscores the significance of testing in our development processes. Eager for more insights in the future! 🚀🔍

**Comment by SamBWagner** [comment]
I think testing is great! Too much testing is not great! No testing is not great!

**Comment by JackDuan123** [comment]
You will never know what testers bring back to you if you work with them.

**Comment by stephanfakossw** [comment]
Amazing video guys! Informative, energetic and hilarious 🙂

**Comment by JackDevAU** [comment]
First up - Luke C and Piers killed it ⭐

Testing is super important, and this series has made me want to write more and more tests. 🚀

**Comment by jernejk** [comment]
I have learned, after watching all of the SSW and other people's testing, everyone has a bit different idea on testing and a bit different definition on what is what, so it's great to define that head of time. 😂

I love to learn all of this but I hope we'll have some actionable items after this, like how we sell testing, how we organise teams to include more testing, plan PBIs with testing in mind, improve testing on internal projects, etc.

The theory is great, let's try to put it into practice. ☺️

**Comment by ozair-ashfaque** [comment]
DTF - Thanks to Piers and Luke C for DTF. 

Testing - TDD approach can improve the quality, reducing the risk of defects and ultimately lowering the overall cost of development.

**Reply by JackDuan123** [reply]
TDD is development techniques, not testing

**Comment by drwharris** [comment]
We need a video on identifying if you have written enough tests

**Comment by jimmidier** [comment]
I learned the way to legitimately view the role of testing, and how to balance the value and cost of testing as a general principle. Good video and great vibe

**Comment by babakamyljanovssw** [comment]
Through this series I've learned the different types of testing, in which scenarios we can use them and it's importance.
BTW amazing video by Luke C and Piers!

**Comment by Aibono1225** [comment]
This series is really great, I've learned about different test definitions and how to apply them in various scenarios.

**Comment by sylhuang** [comment]
I like this Digesting The Fat format. It helps me understand the related CTF better, as well as filter out interesting responses that I myself might have missed in the discussion.

**Comment by KristenHu** [comment]
Amazing video! I like this two-person presentation, and Luke is so funny, lol🤣
This review is also great, it helps me to clarify the whole testing topic better, I expect to see a new series next time!

**Comment by PennyWalker** [comment]
I learned that testing is all about identifying risk and testing accordingly! :)

Really enjoyed the video - funny but informative, really like the graphics on the TV. :)

I wonder if Cookie got any DMs from the design team... 🤣

**Comment by BrookJeynes** [comment]
It's always good to have tests backing up the code you write. It's about reducing the risk of something going wrong.

**Comment by bradystroud** [comment]
My code is perfect and doesn't need tests, but tests are awesome for making sure other devs don't break it

<summary>
<details>
This is a joke
</details>
</summary>

**Comment by leehawkinsssw** [comment]
I hope this testing series has at least opened up the eyes of devs to the various types of testing designed to mitigate different risks. Building test strategies that blend the use of automation with human testing to address the most significant risks is not easy, but everyone should be practising and incrementally improving the testing they do. I am here to help!

**Comment by christoment** [comment]
I think we should do DTF shortly after CTF was done so people can see the outcome of what they have put in on recent CTF 😄 

Testing series is great - love to see SSW encourages people to also put attention into testing :)
